tools/Azure.Mcp.Tools.Extension/src/Commands/CliGenerateCommand.cs

```diff
@@ -0,0 +1,90 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Azure.Mcp.Core.Commands;
+using Azure.Mcp.Tools.Extension.Models;
+using Azure.Mcp.Tools.Extension.Options;
+using Azure.Mcp.Tools.Extension.Services;
+using Microsoft.Extensions.Logging;
+
+namespace Azure.Mcp.Tools.Extension.Commands;
+
+public sealed class CliGenerateCommand(ILogger<CliGenerateCommand> logger) : GlobalCommand<CliGenerateOptions>
+{
+    private const string CommandTitle = "Generate CLI Command";
+    private readonly ILogger<CliGenerateCommand> _logger = logger;
+    private readonly Option<string> _intentOption = ExtensionOptionDefinitions.CliGenerate.Intent;
```

Comment:
rebase with main and use new pattern, new-command.md can help

Reply:
I pushed changes to adapt to the new pattern.